dev/Interop/CameraCaptureUI/CameraCaptureUI/CameraCaptureUIPhotoCaptureSettings.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#include "pch.h"
+#include "CameraCaptureUIPhotoCaptureSettings.h"
+#include "Microsoft.Windows.Media.Capture.CameraCaptureUIPhotoCaptureSettings.g.cpp"
+
+//using namespace winrt::Windows::Media::Capture;
+using namespace winrt::Windows::Foundation;
+
+namespace winrt::Microsoft::Windows::Media::Capture::implementation
+{
+    void CameraCaptureUIPhotoCaptureSettings::validate()
+    {
+        Size const size = CroppedSizeInPixels;
+        Size const aspect = CroppedAspectRatio;
+        auto hasPhotoFixedSizeConstraint = (size.Width != 0) || (size.Height != 0);
+        auto hasPhotoAspectRatioConstraint = (aspect.Width != 0) || (aspect.Height != 0);
+        auto hasPhotoSizeConstraint = hasPhotoFixedSizeConstraint || hasPhotoAspectRatioConstraint;
+
+        if (hasPhotoAspectRatioConstraint && hasPhotoAspectRatioConstraint)
+        {
+            throw hresult_invalid_argument(L"PhotoSettings can't have both a size and aspect ratio specified");
+        }
+        else if (AllowCropping && hasPhotoSizeConstraint)
```

Comment:
I think this check is incorrect, logic should be - If a size constraint is set then cropping must be enabled.

Reply:
Thanks for pointing out, updated!